platform/linux-generic/pktio/socket.c

```diff
@@ -643,6 +643,10 @@ static int sock_mmsg_recv(pktio_entry_t *pktio_entry, int index ODP_UNUSED,
 		uint16_t pkt_len = msgvec[i].msg_len;
 		int ret;
 
+		if (odp_unlikely(msgvec[i].msg_hdr.msg_flags & MSG_TRUNC)) {
+			odp_packet_free(pkt);
```

Comment:
maybe to just set error bit and deliver this packet to application?

Reply:
The odp_pktin_maxlen() API states:
> Maximum frame length in bytes that the packet IO interface can receive.

I interpret this so that the application shouln't see packets which exceed pktin max length.

Reply:
The error bit simply says the packet was truncated to max length. Perhaps the application is only interested in the first few bytes of a packet?

Reply:
The most specific error type an application can see is odp_packet_has_l2_error(), so it cannot trust packet contents anyway.

Reply:
inconsistent with counters.  You do not increase drop counter for pktio and silently drop packet. I would add at least ODP_DBG() so people on debug can see what is happening.

Reply:
Problem is that you cannot increase drop counters manually with socket pktio. The counters are read directly from ethtool or sysfs. ODP_DBG() may be the best option here.